draft-ietf-quic-http.md

```diff
@@ -373,39 +373,34 @@ stream that carried the request that generated the push.  This allows the server
 push to be associated with a request.  Ordering of a PUSH_PROMISE in relation to
 certain parts of the response is important (see Section 8.2.1 of {{!RFC7540}}).
 
-Unlike HTTP/2, the PUSH_PROMISE does not reference a stream; when a server
-fulfills a promise, the stream that carries the stream headers references a Push
-ID.  This allows a server to fulfill promises in the order that best suits its
-needs.
+Unlike HTTP/2, the PUSH_PROMISE does not reference a stream; it contains a Push
+ID. The Push ID uniquely identifies a server push (see {{frame-push-promise}}).
+This allows a server to fulfill promises in the order that best suits its needs.
 
-The server push response is conveyed on a push stream.  A push stream is a
-server-initiated, unidirectional stream.  A push stream includes a header (see
-{{fig-push-stream-header}}) that identifies the PUSH_PROMISE that it fulfills.
-This header consists of a Push ID, encoded as a variable-length integer.  The
-Push ID identifies a server push (see {{frame-push-promise}}).
+When a server later fulfills a promise, the server push response is conveyed on
+a push stream.  A push stream is a server-initiated, unidirectional stream.  A
+push stream always begins with a header (see {{fig-push-stream-header}}) that
+identifies the Push ID of the PUSH_PROMISE that it fulfills, encoded as a
```

Comment:
The paragraph starts by saying fulfilling a promise and ends by saying PUSH_PROMISE, I think I prefer the lowercase version.